activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -104,11 +104,28 @@ def index_name_for_remove(table_name, options = {})
 
       class V5_0 < V5_1
         def create_table(table_name, options = {})
-          if ActiveRecord::Base.connection.adapter_name == "PostgreSQL"
+          connection_name = self.connection.adapter_name
+          if connection_name == "PostgreSQL"
             if options[:id] == :uuid && !options[:default]
               options[:default] = "uuid_generate_v4()"
             end
           end
+
+          # Since 5.1 Postgres adapter uses bigserial type for primary
+          # keys by default and MySQL uses bigint. This compat layer makes old migrations utilize
+          # serial/int type instead -- the way it used to work before 5.1.
```

Comment:
@sgrif Agreed -- that's why I was hoping there was a good default. it sounds like the `else` block works here.  So I'm suggesting something like this:

```ruby
if options[:id].blank?
            if self.connection.class.const_defined?("Compatibility")
              options.merge(self.connection.class::Compatibility::INTEGER_PRIMARY_KEY_OPTIONS)
            else
              options[:id] = :integer
              options[:auto_increment] = true
            end
          end
```